Makefile improvements and better floating point comparisons

reuterbal opened the pull request:
Hi,

thank you for developing libtappp further! I employ it in unit tests for our numeric code, which mainly deals with FLOPs. Better floating point checks were therefore mandatory. 
Let me know if you like the changes.

Best,
Balthasar

reuterbal committed:
Extended floating point comparison functions to deal with small numbers

reuterbal committed:
Makefile does now build sanity test correctly (You should link at the end...)

reuterbal committed:
Makefile cleanup

cbab commented:
Hi Balthasar,

Thank you for submitting these changes! 

I'm fine with the Makefile changes. 

The first patch implementing floating point comparison could be enhanced further:
- Split up/lift up the conditions checking from the ok(...) call.
- The nested condition is the same in the 4 cases (float is()/isnt(), double is()/isnt()), just different min and max threshold comparison. Perhaps this could be refactored in a function where you pass the limits you want to check?
- Could you add a test in the testsuite highlighting this enhanced floating point comparison?

Any further thoughts?

Thank you,

Christian

reuterbal committed:
refactored floating point comparison

reuterbal committed:
Highlighted floating point comparison in test suite

reuterbal committed:
Another test highlighting inequalities in floating-point cases

reuterbal commented:
Hi Christian,

thanks for reviewing the changes and giving feedback!

I refactored the comparison into a new function template, carrying out the nearly-equal-comparison, and added a few more words to the nested comparison along with the link to the floating point guide.

Two new tests are in the testsuite, highlighting nearly-equal floats and float-inequality. 

Best,
Balthasar

reuterbal committed:
Further refactoring of is/isnt

reuterbal committed:
Calls simplified

reuterbal committed:
Added mixed specializations for float/double

reuterbal committed:
Updated sanity tests

cbab commented:
Hi Balthasar,

Thank you for the improved patches. I have pulled in the changes.

Thanks!

Christian
